src/main/java/com/synopsys/integration/alert/channel/jira/web/JiraCustomEndpoint.java

```diff
@@ -79,18 +74,12 @@ public JiraCustomEndpoint(final CustomEndpointManager customEndpointManager, fin
         try {
             final JiraCloudServiceFactory jiraServicesCloudFactory = jiraProperties.createJiraServicesCloudFactory(logger, gson);
             final JiraAppService jiraAppService = jiraServicesCloudFactory.createJiraAppService();
-            final FieldValueModel adminUsername = fieldValueModels.get(JiraGlobalUIConfig.KEY_JIRA_ADMIN_USERNAME);
-            final FieldValueModel adminAccessToken = fieldValueModels.get(JiraGlobalUIConfig.KEY_JIRA_ADMIN_ACCESS_TOKEN);
-            if (null == adminUsername || null == adminAccessToken) {
-                throw new IntegrationException("Both Username and Access Token are required to install plugin.");
-            }
-            final String username = adminUsername.getValue().orElseThrow(() -> new IntegrationException("Expected to be passed an admin username but did not receive any."));
-            final String accessToken = adminAccessToken.getValue().orElseThrow(() -> new IntegrationException("Expected to be passed an admin username but did not receive any."));
+            final String username = jiraProperties.getUsername();
+            final String accessToken = jiraProperties.getAccessToken();
```

Comment:
What error is shown when these are not provided? Shouldn't we still check that these have values?

Reply:
I would expect that if we are missing either of these we would get a 401 from line 79.  We should probably check the error code and if it is unauthorized return that rather than a generic 400 bad request.  For all other status codes then it makes sense to return a 400 Bad Request

Reply:
Paulo is correct, the following call will now handle this error. Additionally, we'd throw an exception in our JiraProperties when trying to connect to the instance.